client/web/src/views/components/content/view-content/chart-view-content/charts/line/helpers/get-processed-chart-data.ts

```diff
@@ -0,0 +1,103 @@
+
+    // In case if we've got unsorted by x (time) axis dataset we have to sort that by ourselves
+    // otherwise we will get an error in calculation of position for the tooltip
+    // Details: bisector from d3-array package expects sorted data otherwise he can't calculate
+    // right index for nearest point on the XYChart.
+    // See https://github.com/airbnb/visx/blob/master/packages/visx-xychart/src/utils/findNearestDatumSingleDimension.ts#L30
+    const sortedData = data.sort((firstDatum, secondDatum) => +accessors.x(firstDatum) - +accessors.x(secondDatum))
+
+    const seriesWithData = series
+        // Separate datum object by series lines
+        .map<LineChartSeriesWithData<Datum>>(line => ({
+            ...line,
+            // Filter select series data from the datum object and process this points array
+            data: getFilteredSeriesData(
+                sortedData.map(datum => ({
+                    x: accessors.x(datum),
+                    y: accessors.y[line.dataKey](datum),
+                }))
+            ),
+        }))
+
+    return { sortedData, seriesWithData }
+}
+
+/**
+ * Filters series data list, preserves null value at the beginning of the series data list
+ * and removes null value between the points.
+ *
+ * Null value ▽   Real point ■                  Null value ▽   Real point ■
+ * ┌────────────────────────────────────┐       ┌────────────────────────────────────┐
+ * │░░░░░░░░░░░░░░░                     │       │░░░░░░░░░░░░░░░                     │
+ * │░░░░░░░░░░░░░░░                     │       │░░░░░░░░░░░░░░░                     │
+ * │░░░░░░░░░░░░░░░                ■    │       │░░░░░░░░░░░░░░░                ■    │
+ * │░░░░░░░░░░░░▽░░    ■                │       │░░░░░░░░░░░░▽░░    ■                │
+ * │░░░░░░░░░░░░░░░          ▽          │──────▶│░░░░░░░░░░░░░░░                     │
+ * │░░░░░░▽░░░░░░░░ ■                   │       │░░░░░░▽░░░░░░░░ ■                   │
+ * │░░░░░░░░░░░░░░░       ■             │       │░░░░░░░░░░░░░░░       ■             │
+ * │░░░▽░░░░░░░░░░░                     │       │░░░▽░░░░░░░░░░░                     │
+ * │░░░░░░░░░░░░░░░             ▽       │       │░░░░░░░░░░░░░░░                     │
+ * └────────────────────────────────────┘       └────────────────────────────────────┘
+ *
+ * @param data - Series data list
+ */
+function getFilteredSeriesData<Datum>(data: Point[]): Point[] {
+    const firstNonNullablePointIndex = Math.max(
+        data.findIndex(datum => datum.y !== null),
+        0
+    )
+
+    // Preserve null values at the beginning of the series data list
+    // but remove null holes between the points further.
+    const nullBeginningValues = data.slice(0, firstNonNullablePointIndex)
+    const pointsWithoutHoles = data
+        // Get values after null area
+        .slice(firstNonNullablePointIndex)
+        .filter(point => point.y !== null)
+
+    return [...nullBeginningValues, ...pointsWithoutHoles]
```

Comment:
Would adding the index condition to filter work here?

```suggestion
    // Preserve null values at the beginning of the series data list
    // but remove null holes between the points further.
    return data.filter((point, index) => point.y !== null || index <= firstNonNullablePointIndex)
```